servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/DefaultHostAndPort.java

```diff
@@ -30,8 +40,71 @@ final class DefaultHostAndPort implements HostAndPort {
      * @param port the port.
      */
     DefaultHostAndPort(String hostName, int port) {
+        this(hostName, port, isIPv6(hostName) ? '[' + hostName + "]:" + port : hostName + ':' + port);
+    }
+
+    private DefaultHostAndPort(String hostName, int port, String toString) {
         this.hostName = requireNonNull(hostName);
         this.port = port;
```

Comment:
Should we check the port range?

Reply:
I tried this but we have some code that depends upon passing in an invalid port to clarify it isn't specified (see `AbstractHttpRequestMetaDataTest` test failures). if we want to change this lets do it in a followup PR.